### PR TITLE
Toolbar priority nav

### PR DESCRIPTION
**Pull request by vxsx**
new things

screen shot 2015-10-01 at 18 26 02

screen shot 2015-10-01 at 18 26 15

screen shot 2015-10-01 at 18 26 29

adds throttle helper to cms.api.helpers
fixes a bug where dragitem couldn't be toggled when clicking on the arrow

### CODE REVIEW

File: cms/static/cms/js/modules/cms.navigation.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+            /**
+             * Cache UI jquery objects
+             * @method _setupUI
+             * @private
+             */
+            _setupUI: function _setupUI() {
+                var trigger = $('.cms-toolbar-more');
+                this.ui = {
+                    window: $(window),
+                    toolbarLeftPart: $('.cms-toolbar-left'),
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: why not using the cms container here as well, similar to modal or sideframe?

File: cms/static/cms/js/modules/cms.navigation.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+                };
+            },
+
+            /**
+             * Setup resize handler to construct the dropdown
+             * @method _events
+             * @private
+             */
+            _events: function _events() {
+                this.ui.window.on('resize.cms load.cms', CMS.API.Helpers.throttle(
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: save events as variables, should probably also be ``resize.cms.navigation``

File: cms/static/cms/js/modules/cms.navigation.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+                    this._handleResize.bind(this), 50
+                ));
+            },
+
+            /**
+             * Calculates all the movable menu items widths
+             * @method _getWidths
+             */
+            getWidths: function getWidths() {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: should this be public?

File: cms/static/cms/js/modules/cms.navigation.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+             */
+            _events: function _events() {
+                this.ui.window.on('resize.cms load.cms', CMS.API.Helpers.throttle(
+                    this._handleResize.bind(this), 50
+                ));
+            },
+
+            /**
+             * Calculates all the movable menu items widths
+             * @method _getWidths
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: add another bullet in-between

File: cms/static/cms/js/modules/cms.navigation.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+                that.items.moreButtonWidth = that.ui.trigger.outerWidth();
+            },
+
+            /**
+             * Calculates available width based on the state of the page
+             *
+             * @method calculateAvailableWidth
+             * @return {Number} available width in px
+             */
+            calculateAvailableWidth: function calculateAvailableWidth() {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: I'd consider adding all those functions to the public space

File: cms/static/cms/js/modules/cms.navigation.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+             * Hides the dropdown
+             *
+             * @method hideDropdown
+             */
+            hideDropdown: function hideDropdown() {
+                this.ui.trigger.css('display', 'none');
+            },
+
+            /**
+             * Figures out if we need to show/hide/modify the dropdown
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: dots ;)

File: cms/static/cms/js/modules/cms.navigation.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+                    if (newRightMostItemIndex < this.rightMostItemIndex) {
+                        this.moveToDropdown(this.rightMostItemIndex - newRightMostItemIndex);
+                    } else if (this.rightMostItemIndex < newRightMostItemIndex) {
+                        this.moveOutOfDropdown(newRightMostItemIndex - this.rightMostItemIndex);
+                    }
+
+                    this.showDropdown();
+
+                    if (remainingWidth < 0 && this.rightMostItemIndex === -1) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.navigation.js]: I would welcome more comments, as this seems a bit complex to me here

File: cms/static/cms/js/modules/cms.navigation.js (hunk 5 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.navigation.js]: what do you mean? it is a public method

File: cms/static/cms/js/modules/cms.navigation.js (hunk 3 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.navigation.js]: don't see why not. if you change the widths of items (idk, change text) you might want to call this again

File: cms/static/cms/js/modules/cms.navigation.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.navigation.js]: ah, well, no reason, will do

File: cms/static/cms/js/modules/cms.navigation.js (hunk 5 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.navigation.js]: and the instance is available from `CMS.API.Toolbar.navigation`

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**